sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/models/QueueMessageDecodingFailure.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.storage.queue.QueueAsyncClient;
+
+import java.util.Objects;
+
+/**
+ * Contains information about message that could not be decoded.
+ */
+@Immutable
+public class QueueMessageDecodingFailure {
+    private final QueueAsyncClient queueAsyncClient;
+    private final QueueMessageItem queueMessageItem;
+    private final PeekedMessageItem peekedMessageItem;
+
+    /**
+     * Creates new {@link QueueMessageDecodingFailure}.
+     * @param queueAsyncClient the {@link QueueAsyncClient} of the queue that has received message.
+     * @param queueMessageItem the {@link QueueMessageItem} that has been received and could not be decoded.
+     * @param peekedMessageItem the {@link PeekedMessageItem} that has been peeked and could not be decoded.
+     */
+    public QueueMessageDecodingFailure(
+        QueueAsyncClient queueAsyncClient,
+        QueueMessageItem queueMessageItem,
+        PeekedMessageItem peekedMessageItem) {
```

Comment:
Consider including the exception that was thrown when decoding here, so the customer can actually take action based on the exception.